app/assets/javascripts/richtext.js

```diff
@@ -1,53 +1,61 @@
-   * the associated preview pne so that it will be regenerated when
-   * the user next switches to it.
-   */
-  $(".richtext_container textarea").change(function () {
-    var container = $(this).closest(".richtext_container");
+(function () {
+  $(document).ready(function () {
+    /*
+    * Install a handler to set the minimum preview pane height
+    * when switching away from an edit pane
+    */
+    $(document).on("hide.bs.tab", ".richtext_container button[data-bs-target$='_edit']", function () {
+      var container = $(this).closest(".richtext_container");
+      var editor = container.find("textarea");
+      var preview = container.find(".tab-pane[id$='_preview']");
+      var minHeight = editor.outerHeight() - preview.outerHeight() + preview.height();
 
-    container.find(".tab-pane[id$='_preview']").empty();
-  }).on("invalid", function () {
-    var container = $(this).closest(".richtext_container");
+      preview.css("min-height", minHeight + "px");
+    });
 
-    container.find("button[data-bs-target$='_edit']").tab("show");
-  });
+    /*
+    * Install a handler to switch to preview mode
+    */
+    $(document).on("show.bs.tab", ".richtext_container button[data-bs-target$='_preview']", function () {
+      var container = $(this).closest(".richtext_container");
+      var editor = container.find("textarea");
+      var preview = container.find(".tab-pane[id$='_preview']");
+
+      if (preview.contents().length === 0) {
+        preview.oneTime(500, "loading", function () {
+          preview.addClass("loading");
+        });
 
-  /*
-   * Install a handler to set the minimum preview pane height
-   * when switching away from an edit pane
-   */
-  $(".richtext_container button[data-bs-target$='_edit']").on("hide.bs.tab", function () {
-    var container = $(this).closest(".richtext_container");
-    var editor = container.find("textarea");
-    var preview = container.find(".tab-pane[id$='_preview']");
-    var minHeight = editor.outerHeight() - preview.outerHeight() + preview.height();
+        preview.load(editor.data("previewUrl"), { text: editor.val() }, function () {
+          preview.stopTime("loading");
+          preview.removeClass("loading");
+        });
+      }
+    });
 
-    preview.css("min-height", minHeight + "px");
+    $(window).on("resize", updateHelp);
   });
 
-  /*
-   * Install a handler to switch to preview mode
-   */
-  $(".richtext_container button[data-bs-target$='_preview']").on("show.bs.tab", function () {
-    var container = $(this).closest(".richtext_container");
-    var editor = container.find("textarea");
-    var preview = container.find(".tab-pane[id$='_preview']");
+  $(document).on("turbo:load", function () {
+    /*
+    * When the text in an edit pane is changed, clear the contents of
+    * the associated preview pne so that it will be regenerated when
+    * the user next switches to it.
+    */
+    $(".richtext_container textarea").change(function () {
```

Comment:
Because `invalid` event doesn't bubble and jQuery, if [this](https://stackoverflow.com/a/24585928) is still true, can't install capture phase listeners. The event will happen at `textarea`, won't go up to `document` and jQuery's delegation won't work.

Installed the listener with `document.addEventListener(..., true)`.

Reply:
Hmm that's unfortunate... Maybe we should leave it in `turbo:load` then? But it would need to be converted to a named function so that it won't get installed more than once on the same element.